tests/core/utilities/test_valid_transaction_params.py

```diff
@@ -38,20 +38,49 @@ def test_assert_valid_transaction_params_invalid_param():
 
 
 def test_extract_valid_transaction_params():
+    ''' baseline test for clean param dict '''
     input = {
         'from': '0x0',
         'to': '0x0',
         'gas': 21000,
         'gasPrice': 5000000,
-        'value': 1,
         'data': '0x0',
+        'value': 1,
         'nonce': 2,
         'chainId': 1,
     }
     valid_transaction_params = extract_valid_transaction_params(input)
     assert valid_transaction_params == input
 
 
+test_data = [({'data': '0x0', 'input': '0x0'}, {'data': '0x0'}), ]
+@pytest.mark.parametrize("transaction_params, expected", test_data)  # noqa E302
+def test_extract_valid_transaction_params_2(transaction_params, expected):
+    valid_transaction_params = extract_valid_transaction_params(transaction_params)
+    assert valid_transaction_params == expected
```

Comment:
parameterize allows us to collapse tests 1-4 into one, like:

```py
# all of these values should probably be different, so we can't accidentally cross-verify values.
FULL_TXN_DICT =  {
        'from': '0x0',
        'to': '0x1',
        'gas': 21000,
        'gasPrice': 5000000,
        'data': '0x2',
        'value': 3,
        'nonce': 2,
        'chainId': 1,
    }


@pytest.mark.parametrize(
    "transaction_params, expected",
    (
        (FULL_TXN_DICT, FULL_TXN_DICT),
        ({'data': '0x0', 'input': '0x0'}, {'data': '0x0'}),
        ({'input': '0x0'}, {'data': '0x0'}),
        ({}, {}),
    ),
)
def test_extract_valid_transaction_params(transaction_params, expected):
    valid_transaction_params = extract_valid_transaction_params(transaction_params)
    assert valid_transaction_params == expected
```

This also nicely sums up the the functionality side-by-side, IMO.